features/share/smart-contracts/README.md

```diff
@@ -0,0 +1,487 @@
+
+Confirm the end result using the Entity Explorer or GraphQL Explorer view of the Share web app.
+
+From the **LenderNode** you should see all 5 loans - 3 with the validation status of `VALID`, 1 with the validation status of `INVALID`, and 1 with the validation status of `ERROR`.
+
+From the **ServicerNode** you should see only 3 loans - the 3 with the validation status of `VALID`.
+
+```graphql
+query ListLoanValidationStatus {
+  list_LoanItems {
+    _LoanItems {
+      ... on Self_Loan {
+        loanIdentifier
+        validationStatus
+      }
+    }
+  }
+}
+```
+
+## Step 3 - Create a Smart Contract for Data Computation
+With several valid loans in place, it's time for the Servicer to update the statistics of its loan portfolio.
+
+### Define Computation To Be Performed
+The Servicer will calculate two values using data from the loans recently added to its loan portfolio.
+
+* **Weighted Average Interest Rate** - The [weighted average](https://www.investopedia.com/terms/w/weightedaverage.asp) of the `interestRatePercent`, weighted by `unpaidPrincipalBalance`
+* **Delinquency Percentage** - The percentage of loans with a `delinquencyStatus` set to `DELINQUENT`
+
+### Create and Configure a Loan Portfolio Statistics Calculation Lambda Function
+**NOTE:** The computation pattern demonstrated here is just one of several that can be used to compute new values.
+
+The [computation Lambda function](src/computation/index.js) includes [Node.js](https://nodejs.dev/) source code that implements the computation of loan portfolio statistics outlined in the previous section, though any [platform and language](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) supported by Lambda will work.
+
+Creating the Lambda function itself is outside the scope of this example.  You can use the [AWS console](https://docs.aws.amazon.com/lambda/latest/dg/getting-started-create-function.html), the [AWS CLI](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-awscli.html), or more [advanced approaches](https://aws.amazon.com/blogs/compute/better-together-aws-sam-and-aws-cdk/).  See [this example](../../../demos/deployment/smart-contracts-with-sam) for a smart contract deployment example that using the [AWS Serverless Application Model (SAM)](https://aws.amazon.com/serverless/sam/).
+
+Once the Lambda function exists, it must be configured to permit a specific node in the Uni (in this case the **ServicerNode**) to invoke it.
+
+1. First you'll need the Smart Contracts role from the **ServicerNode**, which is found in the output of the command below (`ServicerNode.resources.smartContracts.aws_Role`)
+   ```shell
+   share uni get --uni <your_uni_name>
+   ```
+2. Next you'll use the AWS CLI to set two permissions, one that allows the Smart Contract Principal to view the Lambda function's properties and one that allows the principal to invoke the Lambda function.
```

Comment:
Not blocking, but might be helpful to show how to set these values as variables via the `share` CLI and `jq` filtering.